File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/apply_strings_runme.pl
use strict;
use warnings;
use Test::More tests => 4;
BEGIN { use_ok('apply_strings') }
require_ok('apply_strings');

my $TEST_MESSAGE = "A message from target language to the C++ world and back again.";

is(apply_strings::UCharFunction($TEST_MESSAGE), $TEST_MESSAGE, "UCharFunction"); 

is(apply_strings::SCharFunction($TEST_MESSAGE), $TEST_MESSAGE, "SCharFunction");

File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/apply_strings_runme.cs

using System;
using apply_stringsNamespace;

public class apply_strings_runme {

  private static string TEST_MESSAGE = "A message from target language to the C++ world and back again.";

  public static void Main() {
    if (apply_strings.UCharFunction(TEST_MESSAGE) != TEST_MESSAGE) throw new Exception("UCharFunction failed");
    if (apply_strings.SCharFunction(TEST_MESSAGE) != TEST_MESSAGE) throw new Exception("SCharFunction failed");
    SWIGTYPE_p_char pChar = apply_strings.CharFunction(null);
    if (pChar != null) throw new Exception("CharFunction failed");
  }
}

